framework/elements/BaseElement.js

```diff
@@ -0,0 +1,63 @@
+class BaseElement {
+
+    constructor(locator, name) {
+        this.locator = locator;
+        this.name = name;
+    };
+
+    async getElement(){
+        const element = await $(this.locator);
+        return element;
+    }
+
+    async getElements(){
+        const element = await $$(this.locator);
+        return element;
+    }
+    
+    async getLocator() {
+        return this.locator;
+    }
+
+    async getName() {
+        return this.name;
+    }
+
+    async isDisplayed(timeout){
+        let element;
+        try{
+            element = await $(this.locator);
+            return element.waitForDisplayed({timeout})
+        }catch{
+             element = false;
+        }
+        return element;
+    }
+
+    async isEnabled() {
+        const element = await this.getElement();
```

Comment:
don't you need to use the same approach as in isDisplayed to allow to wait until an element is enabled